src/main/java/com/google/api/codegen/transformer/php/PhpSurfaceNamer.java

```diff
@@ -243,10 +243,15 @@ public String injectRandomStringGeneratorCode(String randomString) {
         if (token.equals(InitFieldConfig.RANDOM_TOKEN)) {
           stringParts.add("time()");
         } else {
-          stringParts.add("\"" + token + "\"");
+          stringParts.add(quoted(token));
         }
       }
     }
     return Joiner.on(". ").join(stringParts);
   }
+
+  @Override
+  public String quoted(String text) {
```

Comment:
Please ensure that this will not start putting single quotes where double quotes are expected.

Reply:
I think it should be okay, the only place this function is used in the project is to enquote entity names (ex: `'[PROJECT_ID]'`):

```
InitCodeTransformer.java
...
String entityValue =
  context.getNamer().quoted("[" + Name.from(entityName).toUpperUnderscore() + "]");
```